File: java/src/test/java/problems201_300/LowestCommonAncestorOfABinarySearchTreeTest.java
package problems201_300;

import concepts.TreeNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class LowestCommonAncestorOfABinarySearchTreeTest extends SampledTest {
    @Test
    public void test() {
        for (LowestCommonAncestorOfABinarySearchTree solution
            : LowestCommonAncestorOfABinarySearchTree.values()) {
            for (LowestCommonAncestorOfABinarySearchTreeSample sample
                : getSamples(LowestCommonAncestorOfABinarySearchTreeSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(
                        solution
                            .lowestCommonAncestor(
                                TreeNode.of(sample.input.root),
                                new TreeNode(sample.input.p),
                                new TreeNode(sample.input.q)
                            ).val
                    ).isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems201_300/LowestCommonAncestorOfABinarySearchTreeTest.kt
package problems201_300

import com.google.common.truth.Truth.assertWithMessage
import concepts.TreeNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class LowestCommonAncestorOfABinarySearchTreeTest : SampledTest() {
    @Test
    fun test() =
        LowestCommonAncestorOfABinarySearchTree.entries.forEach { solution ->
            getSamples<Array<LowestCommonAncestorOfABinarySearchTreeSample>>()
                .forEach { (input, output, message) ->
                    assertWithMessage(message)
                        .that(
                            solution
                                .lowestCommonAncestor(
                                    TreeNode.of(*input.root),
                                    TreeNode(input.p),
                                    TreeNode(input.q),
                                )!!
                                .`val`,
                        ).isEqualTo(output)
                }
        }
}
